tests/test_spider.py

```diff
@@ -827,3 +827,18 @@ def test_undefined_parse_method(self):
         exc_msg = "Spider.parse callback is not defined"
         with self.assertRaisesRegex(NotImplementedError, exc_msg):
             spider.parse(resp)
+
+
+class CallbackDepsTest(unittest.TestCase):
+    """Making sure annotations on all non-abstract callbacks can be resolved."""
```

Comment:
Should we mention scrapy-poet here? Maybe even more this to a separate file, e.g. `test_poet`, to test things intended not to break that downstream?